Edit and reorganize documentation

crandmck opened the pull request:
This is a documentation-only PR, with the following goals:

1. Reorganize, eliminate duplication, and prepare for upcoming addition of Rust implementation.
1. Adapt the documentation for display in the CAI doc site (which uses Docusaurus) as well as in GitHub web interface.  Sometimes this calls for small tweaks to formatting or other changes. I'm explicitly NOT optimizing for raw text display (i.e. the markdown source, which you see, for example, when editing an .md file).
1. Clarify and edit to conform with both Adobe Style Guide, the internal CAI style guide, and markdown best practices.

I'm out of time right now, but I will soon open a paired PR in the https://github.com/contentauth/opensource.contentauth.org repo so we can see how the doc in this PR will actually look in the CAI site.

## Reorganize

I deleted a bunch of files in the `/python` dir that were duplicates of those in the root directory.  There's no reason to have duplicate license, contributing, and code of conduct files; none of the other OS repos do.

Also, removed duplicate docs from `/python` (e.g. fully-duplicated README.md and FAQ.md). I moved some of the Python specific info from the main README to the Python README.

Having duplicate information can potentially be confusing and over time can lead to inconsistency if one version is changed and other isn't (or if they're edited differently).

**Questions**:
- What do you think about moving the [section in the main README on C2PA](https://github.com/crandmck/trustmark/tree/main#using-with-c2pa) to a README file in the `/c2pa` directory?
- I left all the installation and quickstart info in the main README, but it could also be moved into the Python README, since it's Python specific.  This might make sense once the Rust impl is added.  What do you think?

## Adapt for display in CAI doc site

- I made links to .md files relative. Links to code files stay absolute.  Relative links will work in Docusaurus as well as GH.
- Use this at the beginning of the FAQ, so the TOC can be injected below the title:
```
<div class="github-only" >
# TrustMark FAQ

</div>
```

The `div` tag is not visible in GitHub, and hides the title from display in the CAI doc site.

## General edits

Content:

- For installation, why would you install using `requirements.txt` or whatever in the Python dir vs from PyPI?  We should say when to use one way vs. the other.
- I know what ONNX is, but what is `ckpt` (in "Models (ckpt and onnx) are no longer packaged in this repo...")
- Should we use Conda in basic steps to run quickstart test, as for GPU setup?
- I was not able to get the JavaScript implementation to work.  I tried both loading the index.html directly in my Chrome browser and running it from a local server.   Here's what my browser console showed:

![Screenshot 2025-03-17 at 9 22 50 PM](https://github.com/user-attachments/assets/c297b6cf-983f-430b-97fc-c46042359a4b)

Formatting and style:

- I removed "Universal Watermarking for Arbitrary Resolution Images" from the main README title, so it's just "TrustMark".  This aligns with general GitHub practice that the main README title is the same as the repo name. Also, it makes the title _very_ long, which doesn't layout well in the doc site.
- For all headings, we use sentence case vs. headline case.  This is Adobe standard and would be consistent with the rest of the docs.  I made this change everywhere.
- I put blank lines after headings (not strictly required, but a markdown best practice; cf. https://www.markdownguide.org/basic-syntax/).

Still TO DO:
- Copy edit of FAQ.

jcollomosse reviewed (changes requested):
Hi Rand, thank you for these documentation improvements.

1. Re: Duplicate of files in Python folder 
Currently the Python folder is a direct image of the PyPi distribution, which is why it contains LICENSE, README, and other files. These two files are parsed and visible on PyPi. We can remove the duplicated CONTRIBUTING and FAQ but I do want to retain:

a) the MIT LICENSE file at that level, perhaps we can symlink to the root of the repo to address duplication 
b) the previously present  information in the README containing academic paper citation and authorship information, which will be visible on PyPi.

2. Modifications to top-level README
You have proposed to remove the arXiv reference to the technical paper that the repo implements from the start of the README, I would keep that citation.  I would prefer for the affiliation information to stay where it is too, as it is common to do this for academic citation but if you feel strongly it should be footnoted I think within the section would be fine rather than right at the bottom of the document where it lacks context.

You proposed moving all of the data schema material that is common to all TrustMark bindings into just the Python folder. I think it is better to keep it where it was, because all languages operate using this schema not just Python.

3. MIT License
You suggest to remove mention of MIT licensing from the README document, although the LICENSE file remains and we remain MIT licensed.  I think it is helpful to retain the explicit note in the README that this repo is MIT licensed but I agree we don’t need to say it twice within that file.

Otherwise I’m happy with the reorganization changes thanks.

Regarding your questions:
1. Q Why would you install via ‘pip install .’ Vs. PyPi.  A: It is common to offer users the option to use PyPi or install from source - this repo provides that option (see above comments re: aligning the python folder with the PyPi distributed archive)
2. Q:What is ckpt? A: This is the PyTorch model file, loaded by the Python code.  The ONNX is used by the JS code. We need both.
3. Q: Conda steps. A: These are ubuntu specific I think it’s better to have keep the quickstart as is, and not conflate that with GPU environment setup which is rarely quick!  pip install . will provide a platform independent install but likely won’t enable GPU without the OS specific env setup.
4. Q: I was not able to get the JavaScript implementation to work.  A: fixed the issue with path literal (now in main).

Formatting etc:
LGTM thank you!

crandmck commented:
Thanks, John, I appreciate your openness to these changes.  I know there are a lot!

Re the duplicated files in the /python directory--now I understand your rationale for this.  As you say, let's see if we can use symlinks or something for some of them to minimize duplication. I also think there may be a way to include content on https://pypi.org/project/trustmark/ from files other than the README.md, but I need to do some investigation (e.g. c2pa-python does this, but I'm trying to figure out how!)

> You have proposed to remove the arXiv reference to the technical paper that the repo implements from the start of the README, I would keep that citation. I would prefer for the affiliation information to stay where it is too, as it is common to do this for academic citation but if you feel strongly it should be footnoted I think within the section would be fine rather than right at the bottom of the document where it lacks context.

The reference is still there in the very first sentence, cf. https://github.com/crandmck/trustmark/blob/main/README.md.  I reformatted it to be a link on the title, vs a "raw" URL.  I did remove "Universal Watermarking for Arbitrary Resolution Images" from the _title_ of the README, though--Is that what you're referring to?

Re the footnotes:
> I would prefer for the affiliation information to stay where it is too, as it is common to do this for academic citation but if you feel strongly it should be footnoted I think within the section would be fine rather than right at the bottom of the document where it lacks context.

The current README has what I _thought_ were intended to be footnotes (i.e. superscript numbers) but were misformatted.  What I did was reformat them to "proper" GHFM footnotes, which automatically displays the note content (i.e. affiliations in this case) at the bottom of the page.  That's what a footnote is ;-). Again, you can see how it would actually work at https://github.com/crandmck/trustmark/blob/main/README.md. Click on the (superscripted) footnotes and you'll automatically scroll down to the content, which will also be highlighted.  There is no way to make the footnote content (i.e. affiliations) display above the bottom of the page.

If that's not acceptable to you, an alternative would be to put the info inline like this:

... by [Tu Bui](https://www.surrey.ac.uk/people/tu-bui) ([DECaDE](https://decade.ac.uk/) Centre for the Decentralized Digital Economy, University of Surrey, UK), [Shruti Agarwal](https://research.adobe.com/person/shruti-agarwal/) ([Adobe Research](https://research.adobe.com/), San Jose, CA.), and [John Collomosse](https://www.collomosse.com) ([DECaDE](https://decade.ac.uk/) Centre for the Decentralized Digital Economy, University of Surrey, UK and [Adobe Research](https://research.adobe.com/), San Jose, CA).

crandmck commented:
>You proposed moving all of the data schema material that is common to all TrustMark bindings into just the Python folder. I think it is better to keep it where it was, because all languages operate using this schema not just Python.

OK, that makes sense--I didn't realize that. I'll move it back.

crandmck commented on README.md:
```diff
@@ -53,75 +93,64 @@ stego = Image.open('images/ufo_240_Q.png').convert('RGB')
+### Payload encoding
 
-## License
+The raw 100 bits break down into D+E+V=100 bits, where D is the protect payload (for example, 61) and E are the error correction parity bits (e.g. 35) and V are the version bits (always four). The version bits comprise two reserved (unused) bits, and two bits encoding an $.
```
@jcollomosse Reminder to fix this part (sentence ending with `$`)

crandmck replied:
Edited accordingly

crandmck commented on js/README.md:
```diff
@@ -1,46 +1,65 @@
-  - Implements error correction and interpretation of binary watermark data.
+This example consists of a simple HTML file, `index.html` that loads two JavaScript files:
+
+- [`tm_watermark.js`](https://github.com/adobe/trustmark/blob/main/js/tm_watermark.js), the core module that handles watermark detection and decoding and defines key functions for processing images and extracting watermark data. `TRUSTMARK_VARIANT` specifies the TrustMark model variant.
```
@jcollomosse 
I searched all the JS (and HTML) code but couldn't find `TRUSTMARK_VARIANT` used anywhere.  Is this still used?

jcollomosse replied:
Please amend to: `modelConfigs` specifies parameters such as the TrustMark model variant.  As provided, the code will check for both Q and P variants.

crandmck replied:
Edited accordingly.

jcollomosse commented:
> I notice that the info on encoding modes is in FOUR places:

The info is in the README, and in the CONFIG instructions.  The README file is duplicated for the reasons we discussed (PyPi) and agreed to keep.  I think we can remove CONFIG from root and only have it the python repo, since it talks about the python configuration method.   

I would prefer to keep the data schema information in both the README and the CONFIG because the former describes the capability and the latter describes how to configure it.

crandmck commented:
OK, I think this should be ready for your final review @jcollomosse .

As discussed, I moved the detailed info on configuring encoding/error-correction mode into [python/CONFIG.md](https://github.com/crandmck/trustmark/blob/main/python/CONFIG.md#error-correction-level) and just left some high-level info in the README [Data schema section](https://github.com/crandmck/trustmark/blob/main/README.md#data-schema) with link to the former.

The python/README.md is a duplicate of top-level README.md, except I adjusted links so they would work.
